### PR TITLE
[#134408063]  Kuss standardize NotFound and TooManyRecords Errors

### PR DESCRIPTION
**Pull request by jonajonlee**
- Fix findOneBy in all drivers to return/throw same values
- Fix getById in all drivers to return/throw same values
- Add throw method to TooManyRecords and NotFound Errors
- Fix unit tests broken by changes

### CODE REVIEW

**Review by mattsmee — APPROVED** [review]
:shipit:

File: lib/couchdb/index.js [hunk 1]
```diff
@@ -25,6 +21,10 @@ const wrapProjection = R.compose(R.merge, R.objOf('projection'))
 
 const getById = R.curry((couch, bucket, id) =>
   Bluebird.resolve(couch.get(bucket, id))
+  .catch(e => {
+    if (e.statusCode != Err.STATUS.NOT_FOUND) throw e
```
Comment (jxbadam) [inline lib/couchdb/index.js]: !== ?

File: lib/error.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 
-
 //eslint-disable-next-line max-len
 //@see http://stackoverflow.com/questions/31089801/extending-error-in-javascript-with-es6-syntax
 
+const { curry } = require('ramda')
```
Comment (jxbadam) [inline lib/error.js]: Do we need to deconstruct? (is there a lot of harm using the whole library)?

Comment (jonajonlee) [reply lib/error.js]: You know, I never really thought of that. I talked to Jason and he said there's really no benefit/disadvantage (if not negligible) to doing either. So I will require the entire R for the sake of readability.

Thoughts @scull7 ?

File: lib/couchdb/index.js [hunk 3]
```diff
@@ -25,6 +21,11 @@ const wrapProjection = R.compose(R.merge, R.objOf('projection'))
 
 const getById = R.curry((couch, bucket, id) =>
   Bluebird.resolve(couch.get(bucket, id))
+  .catch(R.ifElse(
```
Comment (scull7) [inline lib/couchdb/index.js]: :question: 
```javascript
.catch({ statusCode: Err.STATUS.NOT_FOUND }, R.always(undefined))
```

**Review by scull7 — CHANGES_REQUESTED** [review]

**Comment by scull7** [comment]
@jonajonlee you need to increase the code coverage of your patch.

**Review by scull7 — APPROVED** [review]
:shipit:

**Pull request merged**